src/nih_wayfinding/app/scripts/profiles/user-locations-service.js

```diff
@@ -2,17 +2,110 @@
     'use strict';
 
     /* ngInject */
-    function UserLocations(UserLocationsStub) {
+    function UserLocations(UserLocationsStub, localStorageService, ProfileService) {
 
+        var workingLocation = null;
+        /* This is the basic format of a working location
+        {
+            id: 0, // i.e. an integer-based unique identifier
+            type: '', // e.g. cafe, park, school *OPTIONAL*
+            text: '', // e.g. 'Martha's house'
+            img: '', // e.g. 'glyphicon-letter' *OPTIONAL*
+            address: '', // e.g. 'Cathedral of Learning, Pittsburgh' or '1234 Main Street' *OPTIONAL*
+            feature: {} // i.e. geojson feature
+        }
+        */
         var module = {
-            locationsForUser: locationsForUser
+            newLocation: newLocation,
+            workingLocation: workingLocation,
+            locationsForUser: locationsForUser,
+            addLocation: addLocation,
+            removeLocation: removeLocation,
+            getLocationByID: getLocationByID
         };
         return module;
 
-        function locationsForUser(username) {
-            // STUB
-            return UserLocationsStub;
+        /**
+         * Return all locations for a user
+         * @return {array} Array of locations
+         */
+        function locationsForUser() {
+            var user = ProfileService.getCurrentUser();
+            var userLocations = user.locations || [];
+            return UserLocationsStub.concat(userLocations);
+        }
+
+        /**
+         * Create an iterated location ID for a given user
+         * @return {int}
+         */
+        function createLocationID() {
+            var user = ProfileService.getCurrentUser();
+            var locationList = user.locations === 'undefined' ? [] : user.locations;
+            var maxID = _.max(locationList, function(loc) { return loc.id; }).id;
+            var newID = isNaN(maxID) ? 1 : maxID + 1;
+            return newID;
+        }
+
+        /**
+         * Create a base location object with an incremented ID
+         * @return {object}
+         */
+        function newLocation() {
+            var newID = createLocationID();
+            return { id: newID };
+        }
+
+        /**
+         * Get a specific location by id
```

Comment:
On that note, since this method really depends on a user object, shouldn't it be on the Profile service? Or event better, a method on the UserProfile instance